yjit/src/core.rs

```diff
@@ -878,6 +878,7 @@ fn add_block_version(blockref: &BlockRef, cb: &CodeBlock) {
     let version_list = get_or_create_version_list(block.blockid);
 
     version_list.push(blockref.clone());
+    version_list.shrink_to_fit();
```

Comment:
Mildly concerned that `shrink_to_fit` may turn insertion into an O(n) operation because the list has to be resized every time we push...